Create manage_tx.py

rffontenelle opened the pull request:
Draft of script to perform some operations in python-doc organization, mainly to help working on Python version bump.

m-aciek commented on scripts/manage_tx.py:
```diff
@@ -0,0 +1,97 @@
+from transifex.api import transifex_api
+
+# Version number of a new versioned project to create
+new_project_version = '3.11'
```
nit: I wonder if it wouldn't be easier to accept this value from script argument

rffontenelle replied:
Yeah, that was my initial desire when I created it (i.e. when bumping python-newest to 3.12), but I didn't want to expend more time on argparse at the time. Any feedback is welcome

m-aciek replied:
I think [Fire](https://google.github.io/python-fire/) (with version 1 or 4 from [hello world docs](https://google.github.io/python-fire/guide/#version-1-firefire)) should be feasible replacement for argparse that would handle this one extra argument for selected command. Should be more minimal than e.g. Typer or Click which have more verbose API.

rffontenelle replied:
I liked Fire, but it looks like it is not compatible with 3.11 (and because a small fix with PR open and not merged yet). I've came across with Typer and Click before. I guess this could be a good time to play with them.

rffontenelle commented on scripts/manage_tx.py:
```diff
@@ -0,0 +1,97 @@
+        translation_memory_fillup=p.attributes.get('translation_memory_fillup'),
+    )
+    
+    print(f'Creating {vers_proj_name}\'s resources:')
+    for r in RESOURCES:
+        print(f'r.name ...')
+        transifex_api.Resource.create(
+            project=versioned_project,
+            i18n_format=r.i18n_format,
+            slug=r.slug,
+            name=r.name,
+            priority=r.priority,
+            accept_translations=True
+        )
```
Remove this workaround when https://github.com/sphinx-doc/sphinx-intl/pull/90 lands in the source code and gets released.

rffontenelle commented:
Removed locking and unlocking functions because it is much faster to do it in Transifex's web UI by browsing the all resources view, checking the checkbox and then setting them to locked.

Removed resource creating in the new project because `sphinx-intl update-txconfig-resources` and `tx push -s` already do it now.

StanFromIreland commented on scripts/manage_tx.py:
```diff
@@ -0,0 +1,84 @@
+PROJ_SLUG = "python-newest"
+
+
+def _get_tx_token() -> str:
+    """
+    Try to get Transifex API token from some sources.
+    First try TX_TOKEN environment variable, then ~/.transifexrc file.
+    """
+    token = ""
+    if 'TX_TOKEN' in os.environ:
+        token = os.environ['TX_TOKEN']
+    else:
+        config = configparser.ConfigParser()
+        config.read(f"{os.path.expanduser('~')}/.transifexrc")
+        for section in config.sections():
+            if section in ["https://www.transifex.com", "https://app.transifex.com"]:
+                token = config[section]["token"]
+                break
+    return token
```
```suggestion
def _get_tx_token() -> str:
    """
    Try to get Transifex API token from some sources.
    First try TX_TOKEN environment variable, then ~/.transifexrc file.
    """
    if os.path.exists(".tx/api-key"):
        with open(".tx/api-key") as f:
            return f.read()

    config = configparser.ConfigParser()
    config.read(os.path.expanduser("~/.transifexrc"))
    try:
        return config["https://www.transifex.com"]["token"]
    except KeyError:
        pass

    return os.getenv("TX_TOKEN", "")
```
Function from sample workflows script, https://github.com/python-docs-translations/transifex-automations/blob/c1bc8b3317ecf0f35f45b6c42bd5ad36c37e4f90/sample-workflows/transifex-util.py#L94 to check `.tx/api-key` too.

rffontenelle replied:
Thanks. I think I'll add .tx/api-key, but also keep TX_TOKEN as a first option. TX_TOKEN is the official environment variable for Transifex CLI tool, so it requires as API token.
